src/package_test/run_test.rs

```diff
@@ -975,3 +985,158 @@ impl RTest {
         Ok(())
     }
 }
+
+impl CMakeTest {
+    /// Execute the CMake test
+    pub async fn run_test(
+        &self,
+        pkg: &ArchiveIdentifier,
+        path: &Path,
+        prefix: &Path,
+        config: &TestConfiguration,
+    ) -> Result<(), TestError> {
+        let span = tracing::info_span!("Running CMake test");
+        let _guard = span.enter();
+
+        let match_spec = MatchSpec::from_str(
+            format!("{}={}={}", pkg.name, pkg.version, pkg.build_string).as_str(),
+            ParseStrictness::Lenient,
+        )?;
+
+        let dependencies = vec!["cmake".parse().unwrap(), match_spec];
+
+        create_environment(
+            "test",
+            &dependencies,
+            config
+                .host_platform
+                .as_ref()
+                .unwrap_or(&config.current_platform),
+            prefix,
+            &config.channels,
+            &config.tool_configuration,
+            config.channel_priority,
+            config.solve_strategy,
+        )
+        .await
+        .map_err(TestError::TestEnvironmentSetup)?;
+
+        let tmp_dir = tempfile::tempdir()?;
+        let cmake_file = tmp_dir.path().join("CMakeLists.txt");
+
+        let mut cmake_content = String::from("cmake_minimum_required(VERSION 3.15)\nproject(cmake_test)\n\n");
```

Comment:
You can also maybe use: https://cmake.org/cmake/help/latest/manual/cmake.1.html#run-the-find-package-tool